addons/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/config/ChannelParams.java

```diff
@@ -0,0 +1,179 @@
+    private Integer longPressTime;
+
+    public ChannelParams(Channel channel) throws IllegalArgumentException {
+        if (channel == null) {
+            throw new IllegalArgumentException("Channel can't be null");
+        }
+
+        ChannelTypeUID channelTypeUID = channel.getChannelTypeUID();
+        if (channelTypeUID != null) {
+            channelTypeId = channelTypeUID.getId();
+        }
+
+        Map<String, Object> map = channel.getConfiguration().getProperties();
+        for (Map.Entry<String, Object> entry : map.entrySet()) {
+            switch (entry.getKey()) {
+                case PARAM_RESOURCE_ID:
+                    resourceId = getChannelParameterAsInteger(entry.getValue());
+                    break;
+                case PARAM_DIRECTION:
+                    direction = getChannelParameterAsString(entry.getValue());
+                    break;
+                case PARAM_CMD_TO_REACT:
+                    commandToReact = getChannelParameterAsString(entry.getValue());
+                    break;
+                case PARAM_SERIAL_NUMBER:
+                    serialNumber = getChannelParameterAsLong(entry.getValue());
+                    break;
+                case PARAM_PULSE_WIDTH:
+                    pulseWidth = getChannelParameterAsInteger(entry.getValue());
+                    break;
+                case PARAM_LONG_PRESS_TIME:
+                    longPressTime = getChannelParameterAsInteger(entry.getValue());
+                    break;
+                case PARAM_INVERTED:
+                    inverted = getChannelParameterAsBoolean(entry.getValue());
+                    break;
+            }
+        }
+    }
+
+    public String getChannelTypeId() {
+        return channelTypeId;
+    }
+
+    public Integer getResourceId() {
+        return resourceId;
+    }
+
+    public String getDirection() {
+        return direction;
+    }
+
+    public boolean isDirectionBoth() {
+        return direction != null && DIRECTION_READ_WRITE.equals(direction);
+    }
+
+    public boolean isDirectionReadOnly() {
+        return direction != null && DIRECTION_READ_ONLY.equals(direction);
+    }
+
+    public boolean isDirectionWriteOnly() {
+        return direction != null && DIRECTION_WRITE_ONLY.equals(direction);
+    }
+
+    public String getCommandToReact() {
+        return commandToReact;
+    }
+
+    public Integer getPulseWidth() {
+        return pulseWidth;
+    }
+
+    public Boolean getInverted() {
+        return inverted;
+    }
+
+    public boolean isInverted() {
+        if (inverted != null) {
+            return inverted;
+        }
+        return false;
+    }
+
+    public Long getSerialNumber() {
+        return serialNumber;
+    }
+
+    public Integer getLongPressTime() {
+        return longPressTime;
+    }
+
+    private Boolean getChannelParameterAsBoolean(Object value) throws IllegalArgumentException {
+        if (value != null) {
+            try {
```

Comment:
If we want something to keep the exception handling similar, we could do it like this, but it might be a little bit Java 8 oriented:

```java

    private static Boolean getChannelParameterAsBoolean(Object value) throws IllegalArgumentException {
        return convert(value, (v) -> (Boolean) v);
    }
    private Integer getChannelParameterAsInteger(Object value) throws IllegalArgumentException {
        return convert(value, (v) -> ((BigDecimal) v).intValue());
    }
    private Long getChannelParameterAsLong(Object value) throws IllegalArgumentException {
        return convert(value, (v) -> ((BigDecimal) v).longValue());
    }
    private String getChannelParameterAsString(Object value) throws IllegalArgumentException {
        return convert(value, (v) -> (String) v);
    }
    private static <T> T convert(Object value, Function<Object, T> f) {
        if (value == null) {
            return null;
        }

        try {
            return f.apply(value);
        } catch (ClassCastException e) {
            throw new SomeCustomException("Failed to convert", e);
        }
    }

Reply:
Done